File: Data Structures & Algorithms/single-number/submission-0.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        std::unordered_map<int,int> n;
        for (int i : nums) n[i]++;
        
        for (auto pair : n)
        {
            if (pair.second == 1) return pair.first;  
        }
        return 0;
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        f = {}

        for num in nums:
            f[num] = f.get(num, 0) + 1

        print(f)
        
        for key, value in f.items():
            if value > 1:
                return True
        return False